sdks/go/pubsub/eventStore.go

```diff
@@ -60,47 +58,42 @@ func (es *_eventStore) Add(
 func (es _eventStore) List(
 	ctx context.Context,
 	filter model.EventFilter,
-) (<-chan model.Event, <-chan error) {
-	eventChannel := make(chan model.Event, 1000)
-	errChannel := make(chan error, 1)
-
-	go func() {
-		defer close(eventChannel)
-		defer close(errChannel)
-
-		if err := es.db.View(func(txn *badger.Txn) error {
-			sinceTime := new(time.Time)
-			if nil != filter.Since {
-				sinceTime = filter.Since
-			}
-
-			it := txn.NewIterator(badger.DefaultIteratorOptions)
-			defer it.Close()
-			sinceBytes := []byte(es.eventsByTimestampKeyPrefix + sinceTime.Format(sortableRFC3339Nano))
-			for it.Seek(sinceBytes); it.ValidForPrefix([]byte(es.eventsByTimestampKeyPrefix)); it.Next() {
-				item := it.Item()
-				item.Value(func(v []byte) error {
-					event := model.Event{}
-					if err := json.Unmarshal(v, &event); nil != err {
-						return err
-					}
+	eventChannel chan model.Event,
```

Comment:
This has been simplified to be blocking, and write directly to the channel. It's called from a concurrent context that mirrors that channel back anyways, so this is a nice simplification.